hibernate-core/src/main/java/org/hibernate/engine/spi/LoadQueryInfluencers.java

```diff
@@ -52,12 +52,19 @@
 
 	private final EffectiveEntityGraph effectiveEntityGraph = new EffectiveEntityGraph();
 
+	private Boolean readOnly;
+
 	public LoadQueryInfluencers() {
-		this( null );
+		this( null, null );
 	}
 
 	public LoadQueryInfluencers(SessionFactoryImplementor sessionFactory) {
```

Comment:
Perhaps add  `@Deprecated`

Reply:
Again, we have layered design in that null value is absolutely necessary if we don't wanna overwrite the default readOnly value for StatefulPersistenceContext. For that reason, I suggest we don't use `@Deprecated` annotation here. It simply means 'applying readonly as per the StatefulPersistenceContext', though we also have the option (because of the new method) to overwrite the default value.
See the above explanation why Boolean type is chosen.